File: src/main/java/info/skyblond/archivedag/repo/ProtoMetaRepository.java
package info.skyblond.archivedag.repo;

import info.skyblond.archivedag.model.entity.ProtoMetaEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;

public interface ProtoMetaRepository extends JpaRepository<ProtoMetaEntity, String> {
    ProtoMetaEntity findByPrimaryHash(String primaryHash);

    boolean existsByPrimaryHash(String primaryHash);

    boolean existsByPrimaryHashAndSecondaryHash(String primaryHash, String secondaryHash);

    @Modifying
    @Query("update ProtoMetaEntity m set m.mediaType = ?2 where m.primaryHash = ?1")
    void updateMediaType(String primaryHash, String mediaType);

    @Modifying
    void deleteByPrimaryHash(String primaryHash);
}

File: backend/src/main/kotlin/info/skyblond/archivedag/ariteg/repo/ProtoMetaRepository.kt
package info.skyblond.archivedag.ariteg.repo

import info.skyblond.archivedag.ariteg.entity.ProtoMetaEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying
import org.springframework.stereotype.Repository

@Repository
interface ProtoMetaRepository : JpaRepository<ProtoMetaEntity, String> {

    fun findByPrimaryHash(primaryHash: String): ProtoMetaEntity?

    fun existsByPrimaryHash(primaryHash: String): Boolean

    fun existsByPrimaryHashAndSecondaryHash(primaryHash: String, secondaryHash: String): Boolean

    @Modifying
    fun deleteByPrimaryHash(primaryHash: String)
}
